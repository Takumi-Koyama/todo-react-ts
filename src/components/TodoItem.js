//rfce入力後[tab]で自動入力
//アロー関数に書き換え
import React from 'react'
import './TodoItem.css'
// const obj = {
//   name: "hajime",
//   hobby: "game",
//   birthDay: "1993/01/28"
// }
// const { name, hobby } = obj;
    
const TodoItem = ({todo, onClick}) => {
  return (
    <div className="todo-item" onClick={onClick}>
      <div className="todo-item-title">{todo.title}</div>
      <div className="todo-item-description">{todo.description}</div>
    </div>
  )
}

export default TodoItem
