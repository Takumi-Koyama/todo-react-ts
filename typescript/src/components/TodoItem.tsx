import styles from './TodoItem.css';
import React from 'react';
import { Todo } from 'src/model/Todo';

type Props = {
  todo: Todo;
  //関数の場合の書き方(引数なし、戻り値なし)
  onClick?: () => void;
};

export const TodoItem: React.FC<Props> = ({ todo, onClick }) => {
  return (
    <div className={styles.todoItem} onClick={onClick}>
      <div className={styles.todoItemTitle}>{todo.title}</div>
      <div className={styles.todoItemDescription}>{todo.description}</div>
    </div>
  );
};
