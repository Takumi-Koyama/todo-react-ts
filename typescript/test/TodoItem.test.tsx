import { TodoItem } from '../src/components/TodoItem';
import { Todo } from '../src/model/Todo';
//testingライブラリを直接読み込む
import '@testing-library/jest-dom';
//実際に呼び出すことはなくてもReactを読み込む
import React from 'react';
//render,screenをtestinglibraryからとってくる
import { render, screen, cleanup } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

//テストが終わった時に絶対実行される
afterEach(() => {
  //自分でレンダーしたものを綺麗にしてくれる
  return cleanup();
});

describe('TodoItem components', () => {
  it('matches snapshot', () => {
    //テスト用のtodoを用意する
    const todo: Todo = { title: 'test title', description: 'test description' };
    render(<TodoItem todo={todo} />);
    //スナップショットを取ることで見た目が変わっていなことを確認
    const { asFragment } = render(<TodoItem todo={todo} />, {});
    expect(asFragment()).toMatchSnapshot();
    //デバックでhtml形式で確認する console.logと同じ感じ テストを作る用のconsole.log
    // screen.debug();
  });
  //自分が入れた値が正しく表示されるか
  it('render todo', () => {
    const todo: Todo = {
      title: 'test title',
      description: 'test description',
    };
    render(<TodoItem todo={todo} />);
    //testtitleと記載されているテキストが自分のドキュメントに存在するか
    expect(screen.getByText('test title')).toBeInTheDocument();
    expect(screen.getByText('test description')).toBeInTheDocument();
  });
  //clickは親に依存している テスト用のファンクションの作り方
  it('click todo', () => {
    const mockFunction = jest.fn();
    const todo: Todo = {
      title: 'test title',
      description: 'test description',
    };
    render(<TodoItem todo={todo} onClick={mockFunction} />);
    //対象のドキュメントを特定する方法 idがない場合 取得できるドムの親要素という指定
    const todoItem = screen.getByText('test title')
      .parentElement as HTMLElement;
    //クリックしたいHTMLを引数に渡す
    userEvent.click(todoItem);
    //一回だけ呼ばれているか確認
    expect(mockFunction).toHaveBeenCalledTimes(1);
  });
});
