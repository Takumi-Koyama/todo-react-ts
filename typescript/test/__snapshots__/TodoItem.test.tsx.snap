// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem components matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="todoItem"
  >
    <div
      class="todoItemTitle"
    >
      test title
    </div>
    <div
      class="todoItemDescription"
    >
      test description
    </div>
  </div>
</DocumentFragment>
`;
